File: src/2264.largest-3-same-digit-number-in-string.2346/solution.js
/**
 * @param {string} num
 * @return {string}
 */
var largestGoodInteger = function (num) {
    let start = 0;
    let result = '';
    const N = 3;
    while (start + N - 1 < num.length) {
        let end = start + 1;
        while (end < num.length && num[end] === num[start]) {
            end++;
        }
        if (end - start >= N) {
            const sub = num[start].repeat(N);
            if (sub > result) {
                result = sub;
            }
        }
        start = end;
    }
    return result;
};

File: src/2264.largest-3-same-digit-number-in-string.2346/solution.java
class Solution {
    public String largestGoodInteger(String num) {
        String result = "";
        for(int i=0;i<num.length()-2;i++){
            if(num.charAt(i) == num.charAt(i+1) && num.charAt(i+1) == num.charAt(i+2)){
                String sub = num.substring(i,i+3);
                if(sub.compareTo(result)>0){
                    result = sub;
                }

            }
        }
        return result;
    }
}